packages/react-instantsearch-core/src/connectors/__tests__/connectHitInsights.ts

```diff
@@ -1,11 +1,27 @@
-import connect from '../connectHitInsights';
+import connect, { InsightsClient } from '../connectHitInsights';
 
 jest.mock('../../core/createConnector', () => x => x);
 
-function setup() {
+interface MultiIndexContext {
+  context: {
+    ais: {
+      mainTargetedIndex: string;
+    };
+    multiIndexContext: {
+      targetedIndex: string;
+    };
+  };
+}
+
+function setup(): {
+  insightsClient: jest.Mock<any>;
+  props: {
+    insights: InsightsClient;
+  };
+} {
   const insightsClient = jest.fn();
 
-  const createMultiIndexContext = () => ({
+  const createMultiIndexContext = (): MultiIndexContext => ({
```

Comment:
I see no value in having such precise types in tests, because it's much more work to test something which doesn't pass the types, wdyt?

Reply:
Agree.

But if we remove the `@typescript-eslint/explicit-function-return-type` it should not be an issue anymore :innocent: